docs: async-js (#4581)
influxdata/docs-v2: content/influxdb/v2.0/api-guide/client-libraries/nodejs/query.md

   To prevent SQL injection attacks, avoid concatenating unsafe user input with queries.
   {{% /warn %}}

>     ```js

Reviewer: useless change out of scope of this PR, isn't it ?

    const fluxQuery =
      'from(bucket: "YOUR_BUCKET")
        |> range(start: 0)
        |> filter(fn: (r) => r._measurement == "temperature")'
    ```
   Replace *`YOUR_BUCKET`* with the name of your InfluxDB bucket.

4. Use the `iterateRows()` method of the query client to query InfluxDB.
   `iterateRows()` takes a Flux query and returns table as an asynchronous collection.
   The client returns [table](/{{% latest "influxdb" %}}/reference/syntax/annotated-csv/#tables) metadata and rows as an as an AsyncIterable.

  ```js

>   const myQuery = () => {

Reviewer: This should rather be an async lambda with proper error handling inside

   ```
   Replace *`YOUR_BUCKET`* with the name of your InfluxDB bucket.

4. Use the `iterateRows()` method of the query client to query InfluxDB.

>    `iterateRows()` takes a Flux query and returns table as an asynchronous collection.

Reviewer 2: ```suggestion
   `iterateRows()` takes a Flux query and returns the [table](/{{% latest "influxdb" %}}/reference/syntax/annotated-csv/#tables) of metadata and rows as an asynchronous iterable (`AsyncIterable<Row>`).
```

>    The client returns [table](/{{% latest "influxdb" %}}/reference/syntax/annotated-csv/#tables) metadata and rows as an as an AsyncIterable.

Reviewer 2: ```suggestion
   The following example shows how to write an asynchronous function that uses the `iterateRows()` method to query a bucket and uses the JavaScript `for await...of` statement to iterate over the query results:

```

   ```js
   const myQuery = async () => {
     for await (const {values, tableMeta} of queryApi.iterateRows(fluxQuery)) {
       const o = tableMeta.toObject(values)
       console.log(
         `${o._time} ${o._measurement} in '${o.location}' (${o.sensor_id}): ${o._field}=${o._value}`
       )
     }
   }
   myQuery()

>    ```

Reviewer 2: ```suggestion
   ```

To learn more about `for await...of` and async iterable objects, see the [MDN `for await...of` documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of).
```
